ios/RCTMapboxGL/RCTMapboxGLManager.m

```diff
@@ -165,6 +170,82 @@ - (NSDictionary *)constantsToExport
+                [mapView createOfflinePack:bounds style:[NSURL URLWithString:[options valueForKey:@"style"]] fromZoomLevel:[[options valueForKey:@"minZoomLevel"] floatValue] toZoomLevel:[[options valueForKey:@"maxZoomLevel"] floatValue] name:[options valueForKey:@"name"] type:[options valueForKey:@"type"]];
+            } else {
+                RCTLogError(@"Offline type %@ not supported. Only type `bbox` supported.", [options valueForKey:@"type"]);
+            }
+        }
+    }];
+}
+
+RCT_EXPORT_METHOD(getPacksWithCompletionHandler:(nonnull NSNumber *)reactTag
+                  findEvents:(RCTResponseSenderBlock)callback)
+{
+    [_bridge.uiManager addUIBlock:^(RCTUIManager *uiManager, NSDictionary<NSNumber *, RCTMapboxGL *> *viewRegistry) {
+        RCTMapboxGL *mapView = viewRegistry[reactTag];
+        if ([mapView isKindOfClass:[RCTMapboxGL class]]) {
+            RCTMapboxGL *mapView = viewRegistry[reactTag];
+            NSMutableArray* callbackArray = [NSMutableArray new];
+            
+            MGLOfflinePack *packs = [MGLOfflineStorage sharedOfflineStorage].packs;
+            
+            for (MGLOfflinePack *pack in packs) {
+                NSMutableDictionary *packDict = [NSMutableDictionary new];
+                NSDictionary *userInfo = [NSKeyedUnarchiver unarchiveObjectWithData:pack.context];
+                [packDict setObject:userInfo[@"name"] forKey:@"name"];
+                [packDict setObject:@(pack.progress.countOfBytesCompleted) forKey:@"countOfBytesCompleted"];
+                [packDict setObject:@(pack.progress.countOfResourcesCompleted) forKey:@"countOfResourcesCompleted"];
+                [callbackArray addObject:packDict];
+            }
+
+            callback(@[[NSNull null], callbackArray]);
+        }
+    }];
+}
+
+RCT_EXPORT_METHOD(removePack:(nonnull NSNumber *)reactTag
+                  packName:(NSString*)packName
+                  findEvents:(RCTResponseSenderBlock)callback)
+{
+    [_bridge.uiManager addUIBlock:^(RCTUIManager *uiManager, NSDictionary<NSNumber *, RCTMapboxGL *> *viewRegistry) {
+        RCTMapboxGL *mapView = viewRegistry[reactTag];
+        if ([mapView isKindOfClass:[RCTMapboxGL class]]) {
+            RCTMapboxGL *mapView = viewRegistry[reactTag];
+            
+            MGLOfflinePack *packs = [MGLOfflineStorage sharedOfflineStorage].packs;
+            BOOL hasDeletedAPack = NO;
+            
+            for (MGLOfflinePack *pack in packs) {
+                NSDictionary *userInfo = [NSKeyedUnarchiver unarchiveObjectWithData:pack.context];
+                if (userInfo[@"name"] == packName) {
+                    [[MGLOfflineStorage sharedOfflineStorage] removePack:pack withCompletionHandler:^(NSError * _Nullable error) {
+                        if (error != nil) {
+                            return RCTLogError(@"Error: %@", error.localizedFailureReason);
```

Comment:
Remove the return keywords in this block. They just cause the block to return, but the for loop containing the block will continue looping, which is probably not what you intended.